go.mod

```diff
@@ -15,6 +15,7 @@ require (
 	github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
 	github.com/spf13/cobra v0.0.3
 	github.com/spf13/pflag v1.0.3
+	github.com/vmware/go-vmware-nsxt v0.0.0-20190201205556-16aa0443042d // master
```

Comment:
@maplain do you happen to have reference usage of that client for NSX-T?

Reply:
for `https://github.com/vmware/vsphere-automation-sdk-go/tree/master/services/nsxt`, not that I'm aware of. but Ent PKS was using an auto-generated Policy API client: https://gitlab.eng.vmware.com/PKS/pks-networking/tree/master/pkg/client/nsx/policy in https://gitlab.eng.vmware.com/PKS/pks-nsx-t-release/tree/policy_api/src/pkg/networkmanager/policy(though the migration from Manager API to Policy API never happened).

Reply:
Unfortunately I cannot resolve the dns name `gitlab.eng.vmware.com`. We've had a look into this new repo, but to be honest, we couldn't figure out how to use it. Do you have some understandable docu?

Reply:
We found some example and are playing with it, but we still found a severe problem. The elements already created with the *old* API are not visible anymore with the new one.
Therefore is it not possible to setup infrastructure with the old one and switch the controller version to a new controller without deleting the complete cluster and recreating it.
This behavior is basically not acceptable, because we urgently need a valid migration path if VMware evolves its APIs, Assume AWS changes its REST API and suddenly no existing VM is visible anymore. That's definitely a NO-GO.

Reply:
sorry, I lost track of this thread.
May I ask it's the new Policy API or the new Manager API that's lacking some objects? It's ok to go with either one, I brought `https://github.com/vmware/vsphere-automation-sdk-go/tree/master/services/nsxt` up because I'm worried `github.com/vmware/go-vmware-nsxt` will not be maintained anymore.
@andrewsykim do we know if PM has any requirement on the NSX-T version we need to support in CAPV?

Reply:
We've migrated now to the new intent based (automation API). See additional commit from @MartinWeindel